var/spack/repos/builtin/packages/gdal/package.py

```diff
@@ -194,14 +216,26 @@ class Gdal(CMakePackage):
     # Language bindings
     variant("python", default=False, description="Build Python bindings")
     variant("java", default=False, description="Build Java bindings")
-    variant("csharp", default=False, when="@3.5:", description="Build C# bindings")
+    variant("csharp", default=False, when="build_system=cmake", description="Build C# bindings")
     variant("perl", default=False, when="@:3.4", description="Build Perl bindings")
     variant("php", default=False, when="@:2.3", description="Build PHP bindings")
 
+    # Build system
+    build_system(
+        conditional("cmake", when="@3.5:"),
+        conditional("autotools", when="@:3.5"),
+        default="cmake",
+    )
+
+    with when("build_system=cmake"):
+        generator = "Ninja"
```

Comment:
@alalazo does the `generator` parameter belong in Package or Builder? At the moment it's in both in this PR.

Reply:
It should be on the builder.

Reply:
Confirmed that it works as expected.